### PR TITLE
⚡️ Abstracted the field (de)serialization

### PR DESCRIPTION
**Pull request by RenardFute**
> [!CAUTION]
> This pullrequest is based on #36 so please merge this one first

This pull request abstracts the field serialization and de-serialization, so now the component's serialization function is only for custom data that a field is not already taking care of.

> [!WARNING]  
> This may introduce some breaking change if a component was just saving values without field attached.

### CODE REVIEW

File: src/graphics/components/AnimatedSprite.cpp [hunk 1]
```diff
@@ -30,6 +31,13 @@ AnimatedSprite::AnimatedSprite(IObject *owner, const std::string &path,
 
 AnimatedSprite::AnimatedSprite(IObject *owner, const json::JsonObject *data): AComponent(
         owner, new Meta(this), data), frames({}), currentFrame(0), frameTime(0.1f) {
+    this->deserializeFields(data);
+    if (!texture.loadFromFile(path)) {
+        throw GraphicsException("Failed to load texture from file: " + std::string(path));
+        // Todo: add sample texture and only log error
```
Comment (Marius-P1) [inline src/graphics/components/AnimatedSprite.cpp:37]: Did you create an issue for this or inform Aubane ?

File: src/graphics/components/Sprite.cpp [hunk 2]
```diff
@@ -23,6 +24,11 @@ Sprite::Sprite(IObject *owner, const std::string &path) : AComponent(
 
 Sprite::Sprite(IObject *owner, const json::JsonObject *data) : AComponent(
     owner, new Meta(this), data) {
+    this->deserializeFields(data);
+    if (!texture.loadFromFile(path)) {
+        throw GraphicsException("Failed to load texture from file: " + std::string(path));
+    }
```
Comment (Marius-P1) [inline src/graphics/components/Sprite.cpp:30]: I think the ` // Todo: add sample texture and only log error` can be put here too

File: src/graphics/components/SpriteSheet.cpp [hunk 3]
```diff
@@ -29,6 +30,12 @@ SpriteSheet::SpriteSheet(IObject *owner, const std::string &path,
 
 SpriteSheet::SpriteSheet(IObject *owner, const json::JsonObject *data): AComponent(
         owner, new Meta(this), data), frames({}), currentFrame(0) {
+    this->deserializeFields(data);
+    if (!texture.loadFromFile(path)) {
+        throw GraphicsException("Failed to load texture from file: " + std::string(path));
+    }
```
Comment (Marius-P1) [inline src/graphics/components/SpriteSheet.cpp:36]: and here `// Todo: add sample texture and only log error`

File: tests/assets/objects/test-ally.json [hunk 4]
```diff
@@ -9,26 +9,32 @@
...
-        "rotation": {
-          "x": 0.0,
-          "y": 0.0,
-          "z": 0.0
-        },
-        "scale": {
-          "x": 1.0,
-          "y": 1.0,
-          "z": 1.0
+        "invisible": {
```
Comment (Marius-P1) [inline tests/assets/objects/test-ally.json:12]: Why did you add invisible ?

**Review by Marius-P1 — APPROVED** [review]
Same as the other PR, it's ok for me but here we lack code documentation and usage documentation. If we don't read your PR we won't be able to understand why our game don't work anymore and how to update our JSON config file

File: tests/assets/objects/test-ally.json (hunk 4 above)
Comment (RenardFute) [reply tests/assets/objects/test-ally.json:12]: Field are groupped by their field group and it reflects in the json from now on
